### PR TITLE
Fix ansible lint

### PR DESCRIPTION
**Pull request by rndmh3ro**

### CODE REVIEW

**Comment by nre-ableton** [comment]
Thanks for the PR @rndmh3ro, but unfortunately CI is broken due to the role name (though I see that you've already run into this and tried to fix it). I am working on refactoring this role into a collection and will post a proposal soon.

**Comment by rndmh3ro** [comment]
@nre-ableton, I was able to work-around this. It may not be pretty but we can build upon these changes. So please take a look. :)

Nethertheless converting this into a collection is the right way to go.

**Comment by nre-ableton** [comment]
Yes, I see that you got CI working, though I need to review this code before approving/merging. I agree that the solution isn't pretty, but it might be enough to help get us migrated to the collection. I'd prefer if the molecule fixes in particular were short-term. Nonetheless, I'll review this PR, but I'd prefer to merge it after we get some clarity on the migration process (but if that ends up taking too long, we can always merge anyways).

As promised, here's my comment on the proposed collection migration process: https://github.com/ansible-collections/ansible-inclusion/discussions/64#discussioncomment-6962013

File: .ansible-lint [hunk 1]
```diff
@@ -1,26 +1,18 @@
 ---
+exclude_paths:
+  - .github/
 skip_list:
-  # TODO: Fix ansible-lint violations for these errors and remove from skip_list
```
Comment (nre-ableton) [inline .ansible-lint:3]: Please keep this comment here. Some of the items in the skip list (such as `no-handler`) will probably remain forever, but others (like `risky-file-permissions`) should indeed be fixed.

File: .github/workflows/ansible-lint.yml [hunk 2]
```diff
@@ -7,7 +7,6 @@ on:  # yamllint disable-line rule:truthy
   pull_request:
     # The branches below must be a subset of the branches above
     branches: [master]
-
```
Comment (nre-ableton) [inline .github/workflows/ansible-lint.yml:9]: Please keep this newline, unless there is a good reason to get rid of it. The idea was to have separate sections for the name, `on` triggers, and jobs.

File: .github/workflows/molecule.yml [hunk 3]
```diff
@@ -6,7 +6,6 @@ on:
     branches:
       - master
   pull_request:
-
```
Comment (nre-ableton) [inline .github/workflows/molecule.yml]: Please keep this newline, unless there is a good reason to get rid of it. The idea was to have separate sections for the name, `on` triggers, and jobs.

File: .github/workflows/molecule.yml [hunk 4]
```diff
@@ -33,9 +33,6 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v3
 
-      - name: Install test requirements
-        run: pip install -r requirements.txt
-
```
Comment (nre-ableton) [inline .github/workflows/molecule.yml]: So this is actually not necessary? If so, please drop daec0cebd0a066437b3a882f10b949675e3201e5.

**Review by nre-ableton — CHANGES_REQUESTED** [review]
Thanks for the effort here! :+1: 

Regarding the commit history, it would be nice if you could get rid of the merge commit (234d967da16e138a98e1ac90f0dbaba314835057). Can you just rebase that branch into this branch instead?

Please squash all of the commits needed to get CI working (6080d180240996ca506b3f8577cb3cf4f7a9b279, 07c6af46f0a99d431a4c125aaad115d63c557f5c, https://github.com/ansible-community/ansible-consul/pull/574/commits/8efc4e76fad8cd043f430c3d045338398c11614a, and https://github.com/ansible-community/ansible-consul/pull/574/commits/abc9e82d0f13e737ba0bb9b5af216f0d77280c7b)

Also there's a typo in the commit message of `7867ac78c863dbc74dc5c80752730daf294f688d` ("use proper role name and **namespac**")

File: .github/workflows/ansible-lint.yml [hunk 5]
```diff
@@ -1,46 +1,23 @@
 ---
-name: Ansible-lint
+name: Ansible Lint
 
-on:
+on:  # yamllint disable-line rule:truthy
```
Comment (bbaassssiiee) [inline .github/workflows/ansible-lint.yml]: This .yamllint config item takes care of the 'on:', the disable-line is not needed.

 `check-keys: false`

File: .yamllint [hunk 6]
```diff
@@ -10,5 +10,5 @@ rules:
     level: error
   line-length: disable
   truthy:
-    allowed-values: ['true', 'false']
+    allowed-values: ["true", "false"]
```
Comment (bbaassssiiee) [inline .yamllint:13]: Please keep the single quotes for immutable strings

File: molecule/_shared/converge.yml [hunk 7]
```diff
@@ -2,10 +2,10 @@
 - name: Converge
   hosts: all
   roles:
-    - role: ansible-consul
+    - role: brianshumate.consul
```
Comment (bbaassssiiee) [inline molecule/_shared/converge.yml]: Brian is no longer involved since 2020

File: .ansible-lint (hunk 1 above)
Comment (bbaassssiiee) [reply .ansible-lint:3]: You can also annotate the lines with each violations and remove the skip list to avoid new violations creeping in.

**Review by bbaassssiiee — COMMENTED** [review]
It's great that you're involved in fixing technical debt of this role. I'm not sure if all your changes are fixes of problems reported by (your version of) ansible-lint. It seems that you're changing some formatting as well. This role used to have an (implicit) code-style where strings having Jinja2 were double-quoted, and strings having non-alphabetic characters were single-quoted. Would documenting style choices be an idea?

**Review by bbaassssiiee — DISMISSED** [review]

**Comment by nre-ableton** [comment]
> Would documenting style choices be an idea?

Yes, we should do that at some point. If we adopt a quoting style, it should probably be the same one that Ansible uses.

To that end, Ansible is not very clear about quoting style, the only explicit recommendation I've found on the topic is in the [Windows guide](https://docs.ansible.com/ansible/latest/os_guide/windows_usage.html#yaml-style):

> You should only quote strings when it is absolutely necessary or required by YAML, and then use single quotes.

It would be nice if `ansible-lint` enforced this (see https://github.com/ansible/ansible-lint/discussions/3937), but that probably won't happen for awhile.

File: .ansible-lint (hunk 1 above)
Comment (rndmh3ro) [reply .ansible-lint:3]: @bbaassssiiee, good idea, I did that instead of globally excluding them.

File: .github/workflows/ansible-lint.yml (hunk 2 above)
Comment (rndmh3ro) [reply .github/workflows/ansible-lint.yml:9]: I readded it.

File: .github/workflows/molecule.yml (hunk 3 above)
Comment (rndmh3ro) [reply .github/workflows/molecule.yml]: I readded it.

File: .github/workflows/ansible-lint.yml (hunk 5 above)
Comment (rndmh3ro) [reply .github/workflows/ansible-lint.yml]: fixed.

File: molecule/_shared/converge.yml (hunk 7 above)
Comment (rndmh3ro) [reply molecule/_shared/converge.yml]: I changed it to `community` (let's see if it works in github-actions)

**Comment by rndmh3ro** [comment]
>  This role used to have an (implicit) code-style where strings having Jinja2 were double-quoted, and strings having non-alphabetic characters were single-quoted. Would documenting style choices be an idea?

> It would be nice if ansible-lint enforced this (see https://github.com/ansible/ansible-lint/discussions/3937), but that probably won't happen for awhile.

I actually used `ansible-lint --fix` - and it changed the single quotes to double quotes. So I'm in favor of keeping this standard. If ansible-lint decides to use single-quotes or makes it configurable, we can change it IMO.

**Comment by nre-ableton** [comment]
> I actually used `ansible-lint --fix` - and it changed the single quotes to double quotes. So I'm in favor of keeping this standard. If ansible-lint decides to use single-quotes or makes it configurable, we can change it IMO.

Until we have a standard or style guide, I'm personally ok with this approach.

File: molecule/_shared/base.yml [hunk 8]
```diff
@@ -30,10 +30,10 @@ provisioner:
   playbooks:
     prepare: ../_shared/prepare.yml
     converge: ../_shared/converge.yml
+    verify: ../_shared/verify.yml
```
Comment (nre-ableton) [inline molecule/_shared/base.yml:33]: Technically, this commit should also contain the addition of the new `verify.yml` file. However, it's added in a later commit so I won't nitpick about that.

File: molecule/_shared/base.yml [hunk 9]
```diff
@@ -30,10 +30,10 @@ provisioner:
...
   inventory:
     group_vars:
       consul_instances:
         consul_node_role: bootstrap
 verifier:
-  name: testinfra
-  directory: ../_shared/tests
```
Comment (nre-ableton) [inline molecule/_shared/base.yml:39]: I guess that with this PR, the two Python files are obsolete, right? I noticed that in the previous commit you removed the `flake8` verification, which only would have applied to them. If these files are indeed unused, please remove them.

**Review by nre-ableton — COMMENTED** [review]
Looking much better! Thanks for squashing the git history. :+1: 

af95e51d2a2a71d753177b2327fa36fead6d7c81 has a typo in the commit message 🙈 ("make **testss** work...").

**Review by bbaassssiiee — DISMISSED** [review]

**Comment by bbaassssiiee** [comment]
Nice to see that this is in good hands now.

File: molecule/_shared/base.yml (hunk 9 above)
Comment (rndmh3ro) [reply molecule/_shared/base.yml:39]: Thanks, fixed!

**Review by nre-ableton — APPROVED** [review]
Looks great! Very nice work. :+1:

**Pull request merged**